### PR TITLE
Use updatecli to update version

### PR DESCRIPTION
**Pull request by manuelbuil**
Make updatecli update the version

Seems to work: https://github.com/rancher/image-build-cluster-proportional-autoscaler/pull/3

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,4 @@
 ARG GO_IMAGE=rancher/hardened-build-base:v1.20.7b3
-ARG TAG="v1.8.10"
-ARG ARCH="amd64"
 FROM ${GO_IMAGE} as base-builder
```
Comment (rbrtbnfgl) [inline Dockerfile:2]: Can you also remove the `"` from the TAG defined on this file so we don't have an empty PR that only removes the `"`

File: updatecli/updatecli.d/updatednsnodecache.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+name: "Update autoscaler version" 
+
+sources:
+ autoscaler:
+   name: Get autoscaler version
+   kind: githubrelease
+   spec:
+     owner: kubernetes-sigs
+     repository: cluster-proportional-autoscaler
+     token: '{{ requiredEnv .github.token }}'
```
Comment (rbrtbnfgl) [inline updatecli/updatecli.d/updatednsnodecache.yaml:11]:
```suggestion
     token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
```

File: updatecli/updatecli.d/updatednsnodecache.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    spec:
+      file: Makefile
+      matchpattern: '(?m)^TAG \?\= (.*)'
+      replacepattern: 'TAG ?= {{ source "autoscaler" }}$$(BUILD_META)'
+
+scms:
+  default:
+    kind: github
+    spec:
+      token: '{{ requiredEnv .github.token }}'
```
Comment (rbrtbnfgl) [inline updatecli/updatecli.d/updatednsnodecache.yaml:45]:
```suggestion
      token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
```

File: updatecli/updatecli.d/updatednsnodecache.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      username: '{{ requiredEnv .github.username }}'
+      user: '{{ .github.user }}'
```
Comment (rbrtbnfgl) [inline updatecli/updatecli.d/updatednsnodecache.yaml:47]:
```suggestion
      user: '{{  requiredEnv "UPDATECLI_GITHUB_ACTOR" }}'
```

**Review by rbrtbnfgl — CHANGES_REQUESTED** [review]

**Review by macedogm — APPROVED** [review]
LGTM, but there is Roberto's comment to address.

**Review by galal-hussein — APPROVED** [review]

**Review by rbrtbnfgl — APPROVED** [review]

**Pull request merged**